src/java/org/httpkit/ArrayTooLargeException.java

```diff
@@ -0,0 +1,9 @@
+package org.httpkit;
+
+import java.lang.RuntimeException;
+
+public class ArrayTooLargeException extends RuntimeException {
```

Comment:
Maybe this should be an IOException. I'm not a java programmer and I don't know what's best here.

The a checked exception is a mess but may be worth it.